packages/immutable-rootfs/30cos-immutable-rootfs/cos-loop-img.sh

```diff
@@ -9,16 +9,38 @@ function doLoopMount {
         label=$(basename "${dev}")
         [ -e "/tmp/cosloop-${label}" ] && continue
         > "/tmp/cosloop-${label}" 
+
         mount -t auto -o "${cos_root_perm}" "/dev/disk/by-label/${label}" "${cos_state}" || continue
         if [ -f "${cos_state}/${cos_img}" ]; then
-            losetup -f "${cos_state}/${cos_img}"
+
+            # FSCHECK if cos_root_perm == "ro" on both
+            if [ "$cos_root_perm" == "ro" ]; then
+               /usr/lib/systemd/systemd-fsck "/dev/disk/by-label/${label}"
+            fi
+
+            dev=$(losetup --show -f "${cos_state}/${cos_img}")
+
+            # FSCHECK if cos_root_perm == "ro"
+            if [ "$cos_root_perm" == "ro" ]; then
+               /usr/lib/systemd/systemd-fsck "$dev"
+            fi
+
             exit 0
         else
             umount "${cos_state}"
         fi
     done
 }
 
+function dofsCheck {
+    # Iterate over current partitions
```

Comment:
@davidcassany here while testing I corrupted COS_PERSISTENT in a way that no labels were properly displayed, but by running efsck manually I could restore them just fine. 

Since this could happen, in the current initramfs I could not hook into alread-existing loop - so I'm back at scanning all the partitions found. I'm not sure if this is maybe too invasive on some front and maybe better to make it opt-in - but on the other hand I don't see why someone would disable a fscheck integrity utility running on boot. I guess, in that case `fsck.mode=skip` could be passed, so I didn't felt to add any option to instrument it even more.